keras_nlp/models/albert/albert_masked_lm_preprocessor.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Albert masked language model preprocessor layer"""
+
+from absl import logging
+from tensorflow import keras
+
+from keras_nlp.layers.masked_lm_mask_generator import MaskedLMMaskGenerator
+from keras_nlp.models.albert.albert_preprocessor import AlbertPreprocessor
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class AlbertMaskedLMPreprocessor(AlbertPreprocessor):
+    def __init__(
+        self,
+        tokenizer,
+        sequence_length=512,
+        truncate="round_robin",
+        mask_selection_rate=0.15,
+        mask_selection_length=96,
+        mask_token_rate=0.8,
+        random_token_rate=0.1,
+        **kwargs,
+    ):
+        super().__init__(
+            tokenizer,
+            sequence_length=sequence_length,
+            truncate=truncate,
+            **kwargs,
+        )
+
+        self.masker = MaskedLMMaskGenerator(
```

Comment:
@mattdangerw. it seems that masking which the authors of ALBERT followed is slightly different 🤔 . See Section 4.1 of https://arxiv.org/abs/1909.11942.
![image](https://user-images.githubusercontent.com/31360468/216970017-1823dd72-74bd-40f9-ae30-b062f50deee6.png)

Reply:
Thanks! Curious other peoples thought, but I think it's OK if we just do the simple version and expand out over time.

This looks like it is only a preprocessing change correct? The head is still the same?

There's actually a lot of more options we would need to cover--this approach, a "whole word masking" approach used by BERT and other models in follow ups. We can build up these preprocessing options over time I think.

Reply:
Sounds good!